apollo-router/src/state_machine.rs

```diff
@@ -66,6 +68,15 @@ enum State<FA: RouterSuperServiceFactory> {
     Errored(ApolloRouterError),
 }
 
+#[derive(Default)]
+pub(crate) struct LiveReadyState {
+    pub(crate) live: bool,
+    pub(crate) ready: bool,
+}
+
+pub(crate) static LIVE_READY_STATE: Lazy<PLRwLock<LiveReadyState>> =
```

Comment:
I'm really not keen on this mutex.
Let's move this state into the http server factory and have the state machine call methods to mutate the state. 
This will enable us to add unit tests that the state was modified at the right time.